src/test/java/com/amazonaws/encryptionsdk/kms/KMSProviderBuilderIntegrationTests.java

```diff
@@ -115,25 +118,27 @@ public void whenConstructedWithoutArguments_canUseMultipleRegions() throws Excep
         }
     }
 
-    @SuppressWarnings("deprecation") @Test(expected = CannotUnwrapDataKeyException.class)
-    public void whenLegacyConstructorsUsed_multiRegionDecryptIsNotSupported() throws Exception {
+    @SuppressWarnings("deprecation") @Test
+    void whenLegacyConstructorsUsed_multiRegionDecryptIsNotSupported() {
         KmsMasterKeyProvider mkp = new KmsMasterKeyProvider();
 
-        for (String key : KMSTestFixtures.TEST_KEY_IDS) {
-            byte[] ciphertext =
-                    new AwsCrypto().encryptData(
-                            KmsMasterKeyProvider.builder()
-                                                .withKeysForEncryption(key)
-                                                .build(),
-                            new byte[1]
-                    ).getResult();
-
-            new AwsCrypto().decryptData(mkp, ciphertext);
-        }
+        assertThrows(CannotUnwrapDataKeyException.class, () -> {
```

Comment:
This looks to be a bug in the original test, however I believe that the for loop over the keys should be inside the `assertThrows`.

Reply:
Its testing that when decryptData is called on a ciphertext containing a key in the non-default region it fails. The KMSTestFixtures contains one key in the default region and one in the non-default region, so only the second one fails. So the assertThrows needs to be outside the loop since only one key ID is expected to fail